incentivization.md

```diff
@@ -0,0 +1,240 @@
+- the server doesn't have some of the relevant messages and tries to request them from other nodes.
+
+### Server: Incomplete response
+A relevant message is missing from the response.
+Possible explanations:
+- the server didn't receive the message when it was broadcast;
+- the server deliberately withholds the message.
+
+Contrary to blockchains, Relay doesn't have consensus over relayed messages.
+Therefore, it's impossible to distinguish between the two scenarios above.
+TODO: given this fact, what's the best we can aim for?
+
+### Server: Irrelevant response
+The response contains a message that is not relevant.
+There are two scenarios here depending on whether RLN proofs are enforced.
+If RLN is not enforced, a server may insert any number or irrelevant messages into the response.
+If RLN is enforced, a server can only do so as long as it has a valid membership to generate the respective proofs.
+This doesn't eliminate the attack but limits its consequences.
+
+TODO: what are the powers of a malicious server when it comes to generating proofs for irrelevant messages? Can the server generate proofs for past epochs?
+
+### Client: Too many requests
+The client sends many request to the server within a short period of time.
+This may be seen as a DoS attack.
+
+### Client: Request is too large
+The client sends a response that incurs excessive expenses on the server.
+For example, the request covers a very long period in history, or, more generically,
+a period that contains many messages.
+This may also be seen as a DoS attack.
+
+## Privacy considerations
+
+Light protocols, in general, have weaker privacy properties than P2P protocols.
+In a client-server exchange, a client wants to selectively interact with the network.
+By doing so, it often reveals what it is interested in (e.g., subscribes to particular topics).
+
+A malicious Store server can spy on a client in the following ways:
+- track what time frames a client is interested in;
+- analyze the timing of requests;
+- link requests done by the same client.
+
+TODO: expand in the context of an incentivized protocol.
+
+
+# Cost-benefit analysis
+The goal of i13n is to make nodes more likely to exhibit the desired behavior.
+An incentive scheme links the payoffs to whether nodes follow the protocol or not.
+Good behavior should be rewarded, bad behavior punished.
+
+An incentive scheme should balance the costs and benefits for a node.
+Rewards should compensate the cost of good behavior.
+Punishments should offset the benefits that bad behavior may bring.
+
+Let us analyze the costs and benefit of a server that are specific to the Store protocol:
+- storage;
+- bandwidth;
+- computation.
+
+Let us assume a constant flow of messages per epoch and a constant flow of requests for older messages.
+There are two processes: storing incoming messages, and serving old messages to clients.
+
+The cost of storing incoming messages for one epoch is composed of:
```

Comment:
would be nice to have some concrete numbers about the cost, in terms of GB (disk space)

something like https://github.com/waku-org/research/issues/39#issuecomment-1767950815 but properly formulated/expanded.

Reply:
Agree. My preliminary plan is to write everything down in mathematical terms, and then we can plug in concrete numbers, simulate, etc.